pkg/istio-agent/xds_proxy.go

```diff
@@ -130,6 +133,44 @@ func initXdsProxy(ia *Agent) (*XdsProxy, error) {
 		ecdsUpdateChan: make(chan *discovery.DiscoveryResponse, 10),
 	}
 
+	if ia.localDNSServer != nil {
+		proxy.handlers[v3.NameTableType] = func(resp *discovery.DiscoveryResponse) error {
+			if ia.localDNSServer == nil {
+				return nil
+			}
+			if len(resp.Resources) == 0 {
+				return fmt.Errorf("empty response")
+			}
+			var nt nds.NameTable
+			if err := ptypes.UnmarshalAny(resp.Resources[0], &nt); err != nil {
+				log.Errorf("failed to unmarshall name table: %v", err)
+				return err
+			}
+			ia.localDNSServer.UpdateLookupTable(&nt)
+			return nil
+		}
+	}
+	if true { // TODO: add a config option for this
+		proxy.handlers[v3.ProxyConfigType] = func(resp *discovery.DiscoveryResponse) error {
```

Comment:
Move to a function - and comments explaining the plans ( avoid reinjection on proxy config change, etc).